src/or/dirauth/dirvote.c

```diff
@@ -4518,6 +4527,7 @@ dirserv_generate_networkstatus_vote_obj(crypto_pk_t *private_key,
                            options->ConsensusParams, NULL, 0, 0);
     smartlist_sort_strings(v3_out->net_params);
   }
+  v3_out->bwlist_headers = bwlist_headers;
```

Comment:
v3_out->bwlist_headers is set, but never freed. When the vote is freed, you must free both the smartlist and the strings in it.

Reply:
Where is the vote freed?, this sounds like the place https://gitweb.torproject.org/tor.git/tree/src/or/dirauth/dirvote.c#n3616, but not to free bwlist_headers

Reply:
The vote is a networkstatus_t, which is freed by networkstatus_vote_free_ at  https://gitweb.torproject.org/tor.git/tree/src/or/networkstatus.c#n321